core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -819,11 +819,14 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
                 }
 
               if (nextState == TransactionState.PREPARE_ABORT && isEpochFence) {
-                // We should clear the pending state to make way for the transition to PrepareAbort and also bump
-                // the epoch in the transaction metadata we are about to append.
+                // We should clear the pending state to make way for the transition to PrepareAbort
                 txnMetadata.pendingState = None
-                txnMetadata.producerEpoch = producerEpoch
-                txnMetadata.lastProducerEpoch = RecordBatch.NO_PRODUCER_EPOCH
+                if (!clientTransactionVersion.supportsEpochBump()) {
```

Comment:
If I understand correctly, the goal is to keep this logic for TV1 but not use it for TV2. If so lgtm 👍